src/components/visualization/mermaidViz/mermaidViz.tsx

```diff
@@ -0,0 +1,48 @@
+import './mermaidViz.css'
+
+import Stack from '@mui/material/Stack'
+import Typography from '@mui/material/Typography'
+import useTheme from '@mui/system/useTheme'
+import mermaid from 'mermaid'
+import React, { useEffect, useRef } from 'react'
+
+import type { MermaidData } from '../../types'
+
+/** The `MermaidViz` component leverages [Mermaid.js](https://mermaid.js.org/) to create dynamic and interactive diagrams, including flowcharts, sequence diagrams, class diagrams, and more. It is ideal for visualizing complex data and processes in a clear and structured manner. */
+function MermaidViz(props: MermaidData) {
+  const chartRef = useRef(null)
+  const rusticTheme = useTheme()
+  const mermaidTheme = rusticTheme.palette.mode === 'dark' ? 'dark' : 'neutral'
+
+  useEffect(() => {
+    if (chartRef.current) {
+      mermaid.initialize({
+        theme: mermaidTheme,
+        themeVariables: { fontFamily: 'Inter' },
+        ...props.config,
+      })
+
+      mermaid.contentLoaded()
+    }
+  }, [mermaidTheme])
+
+  return (
+    <Stack
+      direction="column"
+      className="rustic-mermaid-container"
+      data-cy="mermaid-container"
+    >
```

Comment:
any particular reason for using Stack instead of a simple div ?

Reply:
I wanted to avoid adding `display: flex; flex-direction: column` by using `Stack`.

Reply:
In a div, elements are rendered one below the other. why do we need flex ?

Reply:
I want mermaid div takes up the remaining space of its parent container. I used `flex:1` in mermaid div and in order to make it work, its parent container needs to have `display: flex`